examples/TransactionGenerator.hs

```diff
@@ -241,178 +256,188 @@ scriptConfigParser = id
                <> "The only commands supported on the commandline are 'poll' and 'listen'.")
-  <*< nodeChainId .:: textOption
-      % long "node-chain-id"
-      <> short 'i'
-      <> metavar "INT"
-      <> help "The specific chain that will receive generated transactions."
+  <*< nodeChainId %:: pLeftSemigroupalUpdate (pure <$> pChainId)
   <*< chainwebHostAddress %:: pHostAddress Nothing
   <*< nodeVersion .:: textOption
       % long "chainweb-version"
       <> short 'v'
       <> metavar "VERSION"
       <> help "Chainweb Version"
+  where
+    pChainId = textOption
+      % long "node-chain-id"
+      <> short 'i'
+      <> metavar "INT"
+      <> help "The specific chain that will receive generated transactions. Can be used multiple times."
 
 data TXGConfig = TXGConfig
-  { _timingdist         :: Maybe TimingDistribution
-  , _genAccountsKeysets :: Map Account (Map ContractName [SomeKeyPair])
-  , _genClientEnv       :: ClientEnv
-  , _genChainId         :: ChainId
-  , _genVersion         :: ChainwebVersion
+  { _confTimingDist :: !(Maybe TimingDistribution)
+  , _confKeysets    :: !(Map Account (Map ContractName [SomeKeyPair]))
+  , _confClientEnv  :: !ClientEnv
+  , _confVersion    :: !ChainwebVersion
   }
 
-makeLenses ''TXGConfig
+confKeysets :: Lens' TXGConfig (Map Account (Map ContractName [SomeKeyPair]))
+confKeysets f c = (\ks -> c { _confKeysets = ks }) <$> f (_confKeysets c)
 
-generateDelay :: (MonadIO m) => TXG m Int
+generateDelay :: MonadIO m => TXG m Int
 generateDelay = do
-  distribution <- view timingdist
-  gen <- use gsGen
+  distribution <- asks _confTimingDist
+  gen <- gets _gsGen
   case distribution of
     Just (Gaussian gmean gvar) -> liftIO (truncate <$> normal gmean gvar gen)
     Just (Uniform ulow uhigh)  -> liftIO (truncate <$> uniformR (ulow, uhigh) gen)
     Nothing                    -> error "generateDelay: impossible"
 
-generateSimpleTransaction :: (MonadIO m, MonadLog SomeLogMessage m) => TXG m (Command Text)
+generateSimpleTransaction
+  :: (MonadIO m, MonadLog SomeLogMessage m)
+  => TXG m (ChainId, Command Text)
 generateSimpleTransaction = do
   delay <- generateDelay
   stdgen <- liftIO newStdGen
   let (operandA, operandB, op) =
         flip evalState stdgen $ do
-            a <- state (randomR (1, 100 :: Integer))
-            b <- state (randomR (1, 100 :: Integer))
-            ind <- state (randomR (0, 2 :: Int))
+            a <- state $ randomR (1, 100 :: Integer)
+            b <- state $ randomR (1, 100 :: Integer)
+            ind <- state $ randomR (0, 2 :: Int)
             let operation = "+-*" !! ind
-            return (a, b, operation)
+            pure (a, b, operation)
       theCode = "(" ++ [op] ++ " " ++ show operandA ++ " " ++ show operandB ++ ")"
-  cid <- view genChainId
+
+  -- Choose a Chain to send this transaction to, and cycle the state.
+  cid <- uses gsChains NES.head
+  gsChains %= rotate
```

Comment:
The meat of this PR. Most other changes are to support this change, as well as fortify logging.

Reply:
Looking good!